Build failure: solc

### Nixpkgs version

- Unstable (26.05)

### Steps to reproduce

`nix build "github:NixOS/nixpkgs/nixos-unstable#solc"`

### Can Hydra reproduce this build failure?

Yes, Hydra can reproduce this build failure.

### Link to Hydra build job

https://hydra.nixos.org/build/318191232

### Relevant log output

```console
In file included from /build/source/libevmasm/Assembly.h:23,
                 from /build/source/libevmasm/Inliner.h:25,
                 from /build/source/libevmasm/Inliner.cpp:23:
In copy constructor 'solidity::evmasm::AssemblyItem::AssemblyItem(const solidity::evmasm::AssemblyItem&)',
    inlined from 'void std::_Construct(_Tp*, _Args&& ...) [with _Tp = solidity::evmasm::AssemblyItem; _Args = {const solidity::evmasm::AssemblyItem&}]' at /nix/store/qarrb8yfby1yyypm32vabzgxgq3w41ma-gcc-15.2.0/include/c++/15.2.0/bits/stl_construct.h:133:7,
    inlined from '_ForwardIterator std::__do_uninit_copy(_InputIterator, _Sentinel, _ForwardIterator) [with _InputIterator = const solidity::evmasm::AssemblyItem*; _Sentinel = const solidity::evmasm::AssemblyItem*; _ForwardIterator = solidity::evmasm::AssemblyItem*]' at /nix/store/qarrb8yfby1yyypm32vabzgxgq3w41ma-gcc-15.2.0/include/c++/15.2.0/bits/stl_uninitialized.h:145:17,
    inlined from '_ForwardIterator std::uninitialized_copy(_InputIterator, _InputIterator, _ForwardIterator) [with _InputIterator = const solidity::evmasm::AssemblyItem*; _ForwardIterator = solidity::evmasm::AssemblyItem*]' at /nix/store/qarrb8yfby1yyypm32vabzgxgq3w41ma-gcc-15.2.0/include/c++/15.2.0/bits/stl_uninitialized.h:266:30,
    inlined from '_ForwardIterator std::__uninitialized_copy_a(_InputIterator, _Sentinel, _ForwardIterator, allocator<_Tp>&) [with _InputIterator = const solidity::evmasm::AssemblyItem*; _Sentinel = const solidity::evmasm::AssemblyItem*; _ForwardIterator = solidity::evmasm::AssemblyItem*; _Tp = solidity::evmasm::AssemblyItem]' at /nix/store/qarrb8yfby1yyypm32vabzgxgq3w41ma-gcc-15.2.0/include/c++/15.2.0/bits/stl_uninitialized.h:637:37,
    inlined from 'void std::vector<_Tp, _Alloc>::_M_range_initialize_n(_Iterator, _Sentinel, size_type) [with _Iterator = const solidity::evmasm::AssemblyItem*; _Sentinel = const solidity::evmasm::AssemblyItem*; _Tp = solidity::evmasm::AssemblyItem; _Alloc = std::allocator<solidity::evmasm::AssemblyItem>]' at /nix/store/qarrb8yfby1yyypm32vabzgxgq3w41ma-gcc-15.2.0/include/c++/15.2.0/bits/stl_vector.h:1989:37,
    inlined from 'std::vector<_Tp, _Alloc>::vector(std::initializer_list<_Tp>, const allocator_type&) [with _Tp = solidity::evmasm::AssemblyItem; _Alloc = std::allocator<solidity::evmasm::AssemblyItem>]' at /nix/store/qarrb8yfby1yyypm32vabzgxgq3w41ma-gcc-15.2.0/include/c++/15.2.0/bits/stl_vector.h:712:23,
    inlined from 'bool solidity::evmasm::Inliner::shouldInlineFullFunctionBody(size_t, ranges::span<const solidity::evmasm::AssemblyItem>, uint64_t) const' at /build/source/libevmasm/Inliner.cpp:159:2:
/build/source/libevmasm/AssemblyItem.h:91:9: error: '<unnamed>[0].solidity::evmasm::AssemblyItem::m_instruction' may be used uninitialized [-Werror=maybe-uninitialized]
   91 |         AssemblyItem(AssemblyItem const&) = default;
      |         ^~~~~~~~~~~~
/build/source/libevmasm/Inliner.cpp: In member function 'bool solidity::evmasm::Inliner::shouldInlineFullFunctionBody(size_t, ranges::span<const solidity::evmasm::AssemblyItem>, uint64_t) const':
/build/source/libevmasm/Inliner.cpp:159:9: note: '<anonymous>' declared here
  159 |         };
      |         ^
In copy constructor 'solidity::evmasm::AssemblyItem::AssemblyItem(const solidity::evmasm::AssemblyItem&)',
    inlined from 'void std::_Construct(_Tp*, _Args&& ...) [with _Tp = solidity::evmasm::AssemblyItem; _Args = {const solidity::evmasm::AssemblyItem&}]' at /nix/store/qarrb8yfby1yyypm32vabzgxgq3w41ma-gcc-15.2.0/include/c++/15.2.0/bits/stl_construct.h:133:7,
    inlined from '_ForwardIterator std::__do_uninit_copy(_InputIterator, _Sentinel, _ForwardIterator) [with _InputIterator = const solidity::evmasm::AssemblyItem*; _Sentinel = const solidity::evmasm::AssemblyItem*; _ForwardIterator = solidity::evmasm::AssemblyItem*]' at /nix/store/qarrb8yfby1yyypm32vabzgxgq3w41ma-gcc-15.2.0/include/c++/15.2.0/bits/stl_uninitialized.h:145:17,
    inlined from '_ForwardIterator std::uninitialized_copy(_InputIterator, _InputIterator, _ForwardIterator) [with _InputIterator = const solidity::evmasm::AssemblyItem*; _ForwardIterator = solidity::evmasm::AssemblyItem*]' at /nix/store/qarrb8yfby1yyypm32vabzgxgq3w41ma-gcc-15.2.0/include/c++/15.2.0/bits/stl_uninitialized.h:266:30,
    inlined from '_ForwardIterator std::__uninitialized_copy_a(_InputIterator, _Sentinel, _ForwardIterator, allocator<_Tp>&) [with _InputIterator = const solidity::evmasm::AssemblyItem*; _Sentinel = const solidity::evmasm::AssemblyItem*; _ForwardIterator = solidity::evmasm::AssemblyItem*; _Tp = solidity::evmasm::AssemblyItem]' at /nix/store/qarrb8yfby1yyypm32vabzgxgq3w41ma-gcc-15.2.0/include/c++/15.2.0/bits/stl_uninitialized.h:637:37,
    inlined from 'void std::vector<_Tp, _Alloc>::_M_range_initialize_n(_Iterator, _Sentinel, size_type) [with _Iterator = const solidity::evmasm::AssemblyItem*; _Sentinel = const solidity::evmasm::AssemblyItem*; _Tp = solidity::evmasm::AssemblyItem; _Alloc = std::allocator<solidity::evmasm::AssemblyItem>]' at /nix/store/qarrb8yfby1yyypm32vabzgxgq3w41ma-gcc-15.2.0/include/c++/15.2.0/bits/stl_vector.h:1989:37,
    inlined from 'std::vector<_Tp, _Alloc>::vector(std::initializer_list<_Tp>, const allocator_type&) [with _Tp = solidity::evmasm::AssemblyItem; _Alloc = std::allocator<solidity::evmasm::AssemblyItem>]' at /nix/store/qarrb8yfby1yyypm32vabzgxgq3w41ma-gcc-15.2.0/include/c++/15.2.0/bits/stl_vector.h:712:23,
    inlined from 'std::optional<solidity::evmasm::AssemblyItem> solidity::evmasm::Inliner::shouldInline(size_t, const solidity::evmasm::AssemblyItem&, const InlinableBlock&) const' at /build/source/libevmasm/Inliner.cpp:226:3:
/build/source/libevmasm/AssemblyItem.h:91:9: error: '<unnamed>[0].solidity::evmasm::AssemblyItem::m_instruction' may be used uninitialized [-Werror=maybe-uninitialized]
   91 |         AssemblyItem(AssemblyItem const&) = default;
      |         ^~~~~~~~~~~~
/build/source/libevmasm/Inliner.cpp: In member function 'std::optional<solidity::evmasm::AssemblyItem> solidity::evmasm::Inliner::shouldInline(size_t, const solidity::evmasm::AssemblyItem&, const InlinableBlock&) const':
/build/source/libevmasm/Inliner.cpp:226:17: note: '<anonymous>' declared here
  226 |                 };
      |                 ^
cc1plus: all warnings being treated as errors
make[2]: *** [libevmasm/CMakeFiles/evmasm.dir/build.make:219: libevmasm/CMakeFiles/evmasm.dir/Inliner.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
make[1]: *** [CMakeFiles/Makefile2:660: libevmasm/CMakeFiles/evmasm.dir/all] Error 2
make: *** [Makefile:136: all] Error 2
```

### Additional context

_No response_

### System metadata

 - system: `"x86_64-linux"`
 - host os: `Linux 6.6.87.2-microsoft-standard-WSL2, Ubuntu, 22.04.5 LTS (Jammy Jellyfish), nobuild`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.23.3`
 - nixpkgs: `/nix/store/yda75qinl8p7pkki117xhqxi6ad36qgx-source`

### Notify maintainers

@dbrock @akru @lionello @sifmelcara 

---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aopen+is%3Aissue+label%3A%220.kind%3A+build+failure%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/by-name/so/solc/package.nix
+++ b/pkgs/by-name/so/solc/package.nix
@@ -25,9 +25,9 @@ assert cvc4Support -> cvc4 != null && cln != null && gmp != null;
 let
   pname = "solc";
 
-  version = "0.8.28";
-  linuxHash = "sha256-kosJ10stylGK5NUtsnMM7I+OfhR40TXPQDvnggOFLLc=";
-  darwinHash = "sha256-gVFbDlPeqiZtVJVFzKrApalubU6CAcd/ZzsscQl22eo=";
+  version = "0.8.33";
+  linuxHash = "sha256-sWCV0GOUW5GPNX1flk+UOrdwoHZHnx4MsZMGDDBxx6M=";
+  darwinHash = "sha256-gyQoBZHOOY1+JyKEa8EOzxd5sToyjvl7aHySzZxwgBo=";
 
   nativeInstallCheckInputs = [
     versionCheckHook
@@ -63,15 +63,6 @@ let
           url = "https://github.com/ethereum/solidity/releases/download/v${version}/solidity_${version}.tar.gz";
           hash = linuxHash;
         };
-
-        # Fix build with GCC 14
-        # Submitted upstream: https://github.com/ethereum/solidity/pull/15685
-        postPatch = ''
-          substituteInPlace test/yulPhaser/Chromosome.cpp \
-            --replace-fail \
-              "BOOST_TEST(abs" \
-              "BOOST_TEST(fabs"
-        '';
 
         cmakeFlags = [
           "-DBoost_USE_STATIC_LIBS=OFF"

